pre_commit/commands/autoupdate.py

```diff
@@ -33,6 +33,20 @@ class RevInfo(NamedTuple):
     def from_config(cls, config: dict[str, Any]) -> RevInfo:
         return cls(config['repo'], config['rev'], None)
 
+    def get_best_candidate_tag(self, rev: str, git_repo: str) -> str:
+        """Get the best tag candidate.
+
+        Multiple tags can exist on a SHA. Sometimes a moving tag is attached
+        to a version tag. Try to pick the tag that looks like a version.
+        """
+        tags = cmd_output(
+            'git', *git.NO_FS_MONITOR, 'tag', '--points-at', rev, cwd=git_repo,
+        )[1].split()
+        for tag in tags:
+            if '.' in tag and re.match(r'^[a-zA-Z0-9.]+$', tag):
```

Comment:
This is, notably, *not* the same as the version detection pattern recommended in #2311. Aside from the `.`, the regex only looks for SHA looking stuff. This pattern looks for a much broader set. I haven't even convinced myself that it's a useful regex to check. I wonder if it would be sufficient to check if the tag has a `.` in it or not.

Reply:
yeah I think `'.' in tag` is probably sufficient (at least for now to be equivalent with the warning) and it can always be improved later 👍